probe.go

```diff
@@ -40,7 +41,8 @@ type Probe struct {
 	kprobeHookPointNotExist bool
 	systemWideID            int
 	programTag              string
-	kprobeType              string
+	kprobeType              probeType
```

Comment:
why not use something generic here like `probeType`? `kprobeType` makes it confusing to read the code.

Reply:
`probeType` is the name of the type, so it gets confusing.

Reply:
`kprobeType` sounds like an instance of a type on the other hand.

Reply:
given there isn't an actual program type of `uprobe` in the kernel (you use `BPF_PROG_TYPE_KPROBE`), I think this is OK for now.